sympy/ntheory/ecm.py

```diff
@@ -182,36 +192,56 @@ def _ecm_one_factor(n, B1=10000, B2=100000, max_curve=200, seed=None):
     ==========
 
     n : Number to be Factored
-    B1 : Stage 1 Bound
-    B2 : Stage 2 Bound
+    B1 : Stage 1 Bound. Must be an even number.
+    B2 : Stage 2 Bound. Must be an even number.
     max_curve : Maximum number of curves generated
 
+    Returns
+    =======
+
+    integer : A non-trivial divisor of ``n``. If ``n`` is prime, then ``n``.
+
+    Raises
+    ======
+
+    ValueError
+        When we cannot find a non-trivial divisor of ``n``.
+
     References
     ==========
 
-    .. [1]  Carl Pomerance and Richard Crandall "Prime Numbers:
-        A Computational Perspective" (2nd Ed.), page 344
+    .. [1] Carl Pomerance, Richard Crandall, Prime Numbers: A Computational Perspective,
+           2nd Edition (2005), page 344, ISBN:978-0387252827
     """
-    n = as_int(n)
-    if B1 % 2 != 0 or B2 % 2 != 0:
-        raise ValueError("The Bounds should be an even integer")
-    sieve.extend(B2)
-
     randint = _randint(seed)
-
     if isprime(n):
         return n
 
-    D = sqrt(B2)
-    beta = [0]*(D + 1)
-    S = [0]*(D + 1)
+    # When calculating T, if (B1 - 2*D) is negative, it cannot be calculated.
+    D = min(sqrt(B2), B1 // 2 - 1)
+    sieve.extend(D)
+    beta = [0] * D
```

Comment:
Was this previously `D+1` so 1-based indexing could be used? If so, and if that matches the indexing of the literature, I am inclined to leave this unchanged since it makes checking/understanding more transparent.

Reply:
OK, thanks. It makes sense as is.